ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/SpacingAroundKeywordRule.kt

```diff
@@ -16,32 +17,44 @@ import org.jetbrains.kotlin.lexer.KtTokens.IF_KEYWORD
 import org.jetbrains.kotlin.lexer.KtTokens.TRY_KEYWORD
 import org.jetbrains.kotlin.lexer.KtTokens.WHEN_KEYWORD
 import org.jetbrains.kotlin.lexer.KtTokens.WHILE_KEYWORD
+import org.jetbrains.kotlin.psi.KtPropertyAccessor
 import org.jetbrains.kotlin.psi.KtWhenEntry
+import org.jetbrains.kotlin.psi.psiUtil.nextLeaf
 
 class SpacingAroundKeywordRule : Rule("keyword-spacing") {
 
     private val noLFBeforeSet = TokenSet.create(ELSE_KEYWORD, CATCH_KEYWORD, FINALLY_KEYWORD)
     private val tokenSet = TokenSet.create(FOR_KEYWORD, IF_KEYWORD, ELSE_KEYWORD, WHILE_KEYWORD, DO_KEYWORD,
         TRY_KEYWORD, CATCH_KEYWORD, FINALLY_KEYWORD, WHEN_KEYWORD)
-    // todo: but not after fun(, get(, set(
+
+    private val keywordsWithoutSpaces = TokenSet.create(KtTokens.GET_KEYWORD, KtTokens.SET_KEYWORD)
 
     override fun visit(node: ASTNode, autoCorrect: Boolean,
-            emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
-        if (tokenSet.contains(node.elementType) && node is LeafPsiElement &&
-            PsiTreeUtil.nextLeaf(node) !is PsiWhiteSpace) {
-            emit(node.startOffset + node.text.length, "Missing spacing after \"${node.text}\"", true)
-            if (autoCorrect) {
-                node.rawInsertAfterMe(PsiWhiteSpaceImpl(" "))
-            }
-        }
-        if (noLFBeforeSet.contains(node.elementType) && node is LeafPsiElement) {
-            val prevLeaf = PsiTreeUtil.prevLeaf(node)
-            if (prevLeaf is PsiWhiteSpaceImpl && prevLeaf.textContains('\n') &&
-                (node.elementType != ELSE_KEYWORD || node.parent !is KtWhenEntry) &&
-                (PsiTreeUtil.prevLeaf(prevLeaf)?.textMatches("}") ?: false)) {
-                emit(node.startOffset, "Unexpected newline before \"${node.text}\"", true)
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+
+        if (node is LeafPsiElement) {
+            if (tokenSet.contains(node.elementType) && node.nextLeaf() !is PsiWhiteSpace) {
+                emit(node.startOffset + node.text.length, "Missing spacing after \"${node.text}\"", true)
                 if (autoCorrect) {
-                    prevLeaf.rawReplaceWithText(" ")
+                    node.rawInsertAfterMe(PsiWhiteSpaceImpl(" "))
+                }
+            } else if (keywordsWithoutSpaces.contains(node.elementType) && node.nextLeaf() is PsiWhiteSpace) {
+                val parent = node.parent
+                if (parent is KtPropertyAccessor && parent.hasBody()) {
+                    emit(node.startOffset, "Unexpected spacing after \"${node.text}\"", true)
+                    if (autoCorrect) {
+                        node.nextLeaf()?.delete()
+                    }
+                }
+            } else if (noLFBeforeSet.contains(node.elementType)) {
```

Comment:
@arturbosch I think we have a regression here. Previously this block would be executed regardless of the preceding if clause (there was no else) (i.e. there could be 2 LintError's emitted for the same node: `Unexpected newline before "else"` & `Missing spacing after "else"`.) but no anymore.